Assets/Scripts/UI/Components/UIButtonLogic.cs

```diff
@@ -29,6 +32,14 @@ void Start()
             {
                 Icon.GetComponent<Image>().color = BaseTextColor;
             }
+           
+                ButtonState(ThisButton.IsInteractable());
+            
+        }
+
+        private void Update()
+        {
+            ButtonState(ThisButton.IsInteractable());
```

Comment:
Kan je hier een check omheen zetten dat de ButtonState alleen getriggered word als IsInteractable gewijzigd is tov de vorige frame? Dat scheelt performance, en zeker met buttons kan het anders oplopen

Reply:
Tevens lees ik in de ButtonState methode dat die GetComponent calls doet; die zijn notoir slecht voor de performance, zeker als het in een update gebeurd. Kan je daar ook "ThisButton" gebruiken?